### PR TITLE
Adds NatSpec to AST for struct definitions.

### LINKED ISSUE #12295: [AST] Consider expanding the use of NatSpec documentation to all declarations.
The motivation behind is, to be able to display additional information about a declaration, not just its type signature.

This might also help documentation.

The relevant AST nodes I'd at least expand to are:

- [x] struct definitions
- [x] enum definitions
- [ ] enum value definitions
- [ ] UDVTs
- [ ] ASTJSON tests for custom errors
- [ ] modules
- [ ] `constant` variable definitions

### PR DESCRIPTION
**Pull request by veniger**
Added NatSpec to AST for struct definitions
Fixes https://github.com/ethereum/solidity/issues/12295

### CODE REVIEW

File: Changelog.md [hunk 1]
```diff
@@ -29,6 +29,7 @@ Language Features:
 
 
 Compiler Features:
+ * NatSpec: Add support for NatSpec documentation in ``struct`` definition.
```
Comment (nikola-matic) [inline Changelog.md]: This is the changelog for the already released 0.8.19 version; you're gonna have to move this up to line 10 (after the enum entry).

**Review by nikola-matic — CHANGES_REQUESTED** [review]

File: Changelog.md [hunk 2]
```diff
@@ -7,6 +7,7 @@ Compiler Features:
  * Assembler: Use ``push0`` for placing ``0`` in the stack for EVM versions starting from "Shanghai". This decreases the deployment costs.
  * EVM: Support for the EVM Version "Shanghai".
  * NatSpec: Add support for NatSpec documentation in ``enum`` definitions.
+ * NatSpec: Add support for NatSpec documentation in ``struct`` definition.
```
Comment (nikola-matic) [inline Changelog.md]:
```suggestion
 * NatSpec: Add support for NatSpec documentation in ``struct`` definitions.
```

File: test/libsolidity/SolidityNatspecJSON.cpp [hunk 3]
```diff
@@ -1076,6 +1076,41 @@ BOOST_AUTO_TEST_CASE(dev_author_at_function)
...
+		"version": 1
+	})ABCDEF";
+
+	checkNatspec(sourceCode, "C", devDoc, false);
+
+	char const* userDoc = R"ABCDEF(
+	{
+	  "kind": "user",
+	  "methods": {},
+	  "version": 1
```
Comment (cameel) [inline test/libsolidity/SolidityNatspecJSON.cpp]: This is indented with a mix of tabs and spaces.

**Review by cameel — COMMENTED** [review]
Looks good other than some minor indent/changelog issues.

BTW, please remember to switch draft PRs to reviewable when they're ready for review.

**Review by nikola-matic — APPROVED** [review]

**Pull request merged**